docs/essentials/parameterized-promotions.md

```diff
@@ -0,0 +1,187 @@
+      env_vars:
+        - required: true
+          options:
+            - Stage
+            - Production
+          default_value: Stage
+          description: Where to deploy?
+          name: ENVIRONMENT
+       
+        - required: false
+          options: []
+          description: Release name?
+          name: RELEASE
+```
+
+## Using the parameter values in promoted pipeline
+Now that we have defined our parameters, it is time to define how they are used in the promoted pipeline. 
+
+Promotion parameters can be used in the following way: 
+
+- In the [pipeline name](https://docs.semaphoreci.com/reference/pipeline-yaml-reference/#example-of-name-usage)
+- In the name of the [pipeline queue](https://docs.semaphoreci.com/reference/pipeline-yaml-reference/#queue)
+- In the name of the [secret](https://docs.semaphoreci.com/reference/pipeline-yaml-reference/#secrets)
+- As an environment variable inside a job
+
+Let's create a deployment pipeline that uses promotion parameter values in all of the mentioned ways: 
+
+``` yaml
+# .semaphore/deploy.yml
+version: v1.0
+# Use parameter values in pipeline name
+name: '${{parameters.ENVIRONMENT}} deployment of the release: ${{parameters.RELEASE}}'
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu2004
+queue:
+  # Use parameter values in pipeline queue name
+  name: '${{parameters.ENVIRONMENT}}-queue'
+  scope: project
+blocks:
+  - name: Param. promotions example
+    task:
+      jobs:
+        - name: Using promotion as env. var
+          commands:
+            # Use parameter values inside a job
+            - echo $ENVIRONMENT
+            - echo $RELEASE
+      secrets:
+        # Use parameter values when attaching a secret
+        - name: 'creds-for-${{parameters.ENVIRONMENT}}'
+```
+**Note:** For this example pipeline to work, secrets named `creds-for-Production` and `creds-for-Stage` have to exist in the organization. 
+
+## Setting the parameter values
+Now that you have configured both the initial pipeline and the promotion pipeline, we can start using parameterized promotions. 
+
+Values to the parameters are assigned every time a promotion is triggered. This can be done in the following ways:
+
+- In the UI form if promotion is triggered manually
+- Through the API call 
+- Through `default_value` in combination with `auto_promote`
+
+### Setting the values through the UI
+In the Semaphore web interface, you will see a deployment button. 
+
+When clicked, a UI form will be auto-generated based on the configuration in the initial pipeline, prompting you for parameter values before starting the promotion. 
+
+Here's what it looks like for the `.yml` code from our example:
+
+<img style="box-shadow: 0px 0px 5px #ccc" src="/essentials/img/parameterized-promotions/ui-form.png" alt="Tests Tab on Workflow Page">
+
+### Setting the values through the API
+Parameters can also be passed to the promotion pipeline when the promotions are triggered through Semaphore API. 
```

Comment:
You can also pass parameters to the promotion pipeline when promotions are triggered via the Semaphore API.